docs/design/service-modules.md

```diff
@@ -0,0 +1,233 @@
+# Service modules architecture
+
+All services are seperated in an api module and an implementation module.
+The name pattern for the api modules is `hedera-node/hedera-[NAME]-service` and the name pattern
+for the modules that contain the implementation is `hedera-node/hedera-[NAME]-service-impl`.
+The modules are all support the Java module system.
+In the following sample `'Foo'`/`'foo'` is used as placeholder for the service name / type.
+
+## Service api modules
+
+All service api modules are based on the same structure and a minimal module looks like this:
+
+```
+hedera-foo-service/
+├── src/main/java/
+│   ├── com.hedera.node.app.service.foo
+│   │   ├── FooService.java
+│   │   └── package-info.java
+│   └── module-info.java
+└── build.gradle.kts
+```
+
+The api modules should depend on the `com.hedera.node.app.spi` module and provide it as a transitive dependency.
+To do so `api(project(":hedera-node:hedera-app-spi"))` must be a dependency in the gradle file.
+The module should only contain the public api of services.
+The module should only contain interfaces, annotations, enums and records. The only classes should be exceptions.
```

Comment:
Not sure, if I can agree to this. With the introduction of default methods, the line between abstract classes and interfaces has become blurry anyway. I think, there are valid use-cases, where you want to include a simple POJO in your API.

Reply:
yes, maybe that is too hard ;)